### PR TITLE
Update qulice to version 0.17.1

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #1120 . 
Fixed newly found errors.

I escaped those which were already escaped before (e.g. PMD now also checks for duplicated strings and there was already an escape for checkstyle, so I added escape for PMD as well). 

I also added escapes where there were already todo puzzles (e.g. ``NbRule`` and ``Ports`` classes)

### CODE REVIEW

**Comment by karato** [comment]
@amihaiemil Thanks, I will find someone to review this PR soon

**Comment by karato** [comment]
@mkordas review it

**Comment by mkordas** [comment]
@amihaiemil I'll try to provide rapid feedback for you here j

**Comment by mkordas** [comment]
@amihaiemil I'm on it - let's finish and merge it today :)

File: netbout-web/src/main/java/com/netbout/rest/RsPage.java [hunk 1]
```diff
@@ -54,6 +54,7 @@
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
 @EqualsAndHashCode(callSuper = true)
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
```
Comment (mkordas) [inline netbout-web/src/main/java/com/netbout/rest/RsPage.java]: @amihaiemil can you remove all suppressions for `@checkstyle MultipleStringLiteralsCheck`? They are not needed anymore

File: netbout-web/src/main/java/com/netbout/rest/TkApp.java [hunk 2]
```diff
@@ -74,7 +74,7 @@
  * @checkstyle MultipleStringLiteralsCheck (500 lines)
```
Comment (mkordas) [inline netbout-web/src/main/java/com/netbout/rest/TkApp.java]: @amihaiemil here and in all other places as well

**Comment by mkordas** [comment]
@amihaiemil also let's remove all `@SuppressWarnings("PMD.DoNotUseThreads")` from the project, as they are not needed anymore

**Comment by mkordas** [comment]
@amihaiemil just two things from me

**Comment by amihaiemil** [comment]
@mkordas done

**Comment by mkordas** [comment]
@amihaiemil looks great, thank you!

**Comment by mkordas** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor please deploy

**Comment by ypshenychka** [comment]
@amihaiemil 
>There is already a todo puzzle in this class to redesign it so it has no public static methods

>This I added because NonStaticMethodCheck is in conflict with No public static methods rule, in this case. It will be solved with the todo that is in this class already. If it wasn't for this conflict, the todo could have been removed.

According to our [QA Rules](http://at.teamed.io/qa.html):
>Messages in a ticket always start with a name of a user they are addressed to.

Please correct your messages by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@mkordas 
According to our [QA Rules](http://at.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by mkordas** [comment]
@ypshenychka I confirm

**Comment by amihaiemil** [comment]
@ypshenychka done

**Comment by ypshenychka** [comment]
@mkordas @amihaiemil Thank you!

**Comment by ypshenychka** [comment]
@karato Quality is acceptable.

**Comment by karato** [comment]
> @karato Quality is acceptable.

@ypshenychka thanks, got it. everybody, please try to make it good next time

**Comment by karato** [comment]
@mkordas 10 mins was added to the account of @ypshenychka (for QA review), in transaction 99408049... I added **24 mins** to your account, many thanks for working with the project (57cddaed57889e1a2cbca6ab)! The completion time here was 19 hours and 58 mins.... review comments (c=9) added as a bonus... +24 to your rating, your total score is [+4328](http://www.netbout.com/b/36080?open=rating)

**Pull request merged**